test/e2e/volume_provisioning.go

```diff
@@ -125,6 +125,41 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 
 			testDynamicProvisioning(c, claim)
 		})
+
+		It("should test the persistent volume is deleted when the claim is deleted. [Slow][Volume]", func() {
+			// This case tests for the regressions of a bug fixed by PR #21268
+			// REGRESSION: Deleting the PVC before the PV is provisioned can result in PV
+			// not being deleted
+			By("creating a StorageClass")
+			class := newStorageClass()
+			_, err := c.Storage().StorageClasses().Create(class)
+			defer c.Storage().StorageClasses().Delete(class.Name, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating and then deleting the claim")
+			claim := newClaim(ns, false)
+			claim, err = c.Core().PersistentVolumeClaims(ns).Create(claim)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for the claim to create before deleting it.
+			err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimPending, c, ns, claim.Name, 1*time.Second, 60*time.Second)
```

Comment:
The pvc while sit in Pending until the volume is provisioned, then be Bound by the controller.  The idea here was just to wait until the claim is pending before deleting it, but I don't think it's actually necessary at all.  Will remove it.